### PR TITLE
Download and pin upstream linux kernel via nix

### PR DESCRIPTION
**Pull request by tiziano88**
First step towards attempting to build and cache it via nix too, but this is also useful in itself to make sure that the archive does not change.

b/328294742

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -25,6 +25,36 @@
...
+          linux_kernel_upstream = builtins.fetchurl {
+            url = "https://cdn.kernel.org/pub/linux/kernel/v6.x/linux-6.7.6.tar.xz";
+            sha256 = "1lrp7pwnxnqyy8c2l4n4nz997039gbnssrfm8ss8kl3h2c7fr2g4";
+          };
+          # TODO: b/328294742 - This derivation does not currently compile correctly, but it is a starting point.
+          oak_containers_kernel = pkgs.stdenv.mkDerivation {
+            name = "oak_containers_kernel";
+            src = ./oak_containers_kernel;
+            buildInputs = with pkgs; [
+              cowsay
```
Comment (andrisaar) [inline flake.nix:37]: what?

File: flake.nix [hunk 2]
```diff
@@ -25,6 +25,36 @@
...
+              elfutils
+              glibc
+              glibc.static
+              cpio
+            ];
+            baseInputs = [
+              linux_kernel_upstream
+            ];
+            buildPhase = ''
+              make target/vmlinux
```
Comment (andrisaar) [inline flake.nix:52]: We want the bzImage, not vmlinux.

**Review by andrisaar — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 1 above)
Comment (tiziano88) [reply flake.nix:37]: debugging by cowsay :) this derivation is not used at all yet, but I wanted to figure out how things connect together

File: flake.nix (hunk 2 above)
Comment (tiziano88) [reply flake.nix:52]: this is also still unused, but it is a starting point if anyone wants to tweak and try things out.

**Review by andrisaar — APPROVED** [review]

**Comment by tiziano88** [comment]
To clarify, the only thing this PR actually changes in terms of functioning logic is that the archive is downloaded and pinned via nix instead of downloaded via curl. That is achieved by the fetchurl stanza. The mkDerivation stanza has no effect at the moment, it's only for future reference.

**Pull request merged**